File: biketest.py
import unittest

class TestBikeFunctions(unittest.TestCase):

    def test_gear_one_adds_one(self):
        self.assertEqual(gear_one(15), 16)
        self.assertEqual(gear_one(0), 1)
        self.assertEqual(gear_one(-5), -4)

    def test_accelerate_valid_gear_1(self):
        try:
            accelerate(10, 1)
        except Exception:
            self.fail("accelerate() raised Exception unexpectedly for gear 1")

    def test_accelerate_valid_gear_4(self):
        try:
            accelerate(50, 4)
        except Exception:
            self.fail("accelerate() raised Exception unexpectedly for gear 4")

    def test_accelerate_invalid_gear_above_4(self):
        with self.assertRaises(ValueError) as context:
            accelerate(25, 5)
        self.assertEqual(str(context.exception), "Invalid gear: must be between 1 and 4")

    def test_accelerate_invalid_gear_below_1(self):
        with self.assertRaises(ValueError) as context:
            accelerate(25, 0)
        self.assertEqual(str(context.exception), "Invalid gear: must be between 1 and 4")

if __name__ == "__main__":
    unittest.main()

File: biketest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class BikeTest {

    @Test
    public void testGearOneAddsOneToSpeed() {
        double result = Bike.gearOne(15.0);
        assertEquals(16.0, result);
    }

    @Test
    public void testAccelerateWithValidGear() {
        // Capture output using a PrintStream if needed, but here we test logic
        assertDoesNotThrow(() -> Bike.accelerate(20.0, 2));
    }

    @Test
    public void testAccelerateWithGearOne() {
        double speed = 10.0;
        int gear = 1;
        double expected = speed + gear;
        Bike.accelerate(speed, gear); // Should print 11.0
        // You can redirect System.out to capture output if needed
    }

    @Test
    public void testAccelerateThrowsExceptionForInvalidGearAbove4() {
        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
            Bike.accelerate(25.0, 5);
        });
        assertEquals("Invalid gear: must be between 1 and 4", exception.getMessage());
    }

    @Test
    public void testAccelerateThrowsExceptionForInvalidGearBelow1() {
        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
            Bike.accelerate(25.0, 0);
        });
        assertEquals("Invalid gear: must be between 1 and 4", exception.getMessage());
    }
}
